docs/tutorials/_static/geomesa-examples-jupyter/shallow-join-gdelt.ipynb

```diff
@@ -0,0 +1,814 @@
+    "    if (typeA == \"countryInformation\" && typeB == \"countryInformation\") {\n",
+    "        // Combine the \"total\" properties\n",
+    "        (featureA.getProperties, featureB.getProperties).zipped.foreach((propA, propB) => {\n",
+    "            val name = propA.getName.toString\n",
+    "            if (propA.getName.toString.startsWith(\"total_\") || propA.getName.toString == \"count\") {  \n",
+    "                val sum = (propA.getValue, propB.getValue) match {\n",
+    "                    case (a: Integer, b: Integer) => a + b\n",
+    "                    case (a: java.lang.Long, b: java.lang.Long) => a + b\n",
+    "                    case (a: java.lang.Double, b: java.lang.Double) => a + b\n",
+    "                    case _ => throw new Exception(\"Couldn't match countable type.\")\n",
+    "                }\n",
+    "                featureA.setAttribute(propA.getName, sum)             \n",
+    "            }\n",
+    "        })\n",
+    "        featureA\n",
+    "        \n",
+    "    // Case: combining two gdelt features\n",
+    "    } else if (typeA != \"countryInformation\" && typeB != \"countryInformation\") {\n",
+    "        \n",
+    "        // Grab each feature's properties\n",
+    "        val featurePropertiesA = featureA.getProperties.toSeq\n",
+    "        val featurePropertiesB = featureB.getProperties.toSeq\n",
+    "        // Create a new aggregate feature to hold the result\n",
+    "        val featureFields = Seq(\"empty\", featureA.geometry) ++ Seq.fill(aggregateSft.getTypes.size-2)(\"0\")                        \n",
+    "        val aggregateFeature = ScalaSimpleFeatureFactory.buildFeature(aggregateSft, featureFields, featureA.getID)                      \n",
+    "        \n",
+    "        // Loop over the countable properties and sum them for both gdelt simple features\n",
+    "        countable.value.foreach { case (index, clazz) =>\n",
+    "            val propA = featurePropertiesA(index)\n",
+    "            val propB = featurePropertiesB(index)\n",
+    "            val valA = if (propA == null) 0 else propA.getValue\n",
+    "            val valB = if (propB == null) 0 else propB.getValue\n",
+    "                                 \n",
+    "            val sum  = (valA, valB) match {\n",
+    "                case (a: Integer, b: Integer) => a + b\n",
+    "                case (a: java.lang.Long, b: java.lang.Long) => a + b\n",
+    "                case (a: java.lang.Double, b: java.lang.Double) => a + b\n",
+    "                case _ => throw new Exception(\"Couldn't match countable type.\")\n",
+    "            }\n",
+    "            // Set the total\n",
+    "            if( propA != null)\n",
+    "                aggregateFeature.setAttribute(\"total_\"+ propA.getName.toString, sum)                      \n",
+    "                  \n",
+    "        }\n",
+    "        aggregateFeature.setAttribute(\"count\", new Integer(2))              \n",
```

Comment:
[picky] If this code is going to be called many times, it may be more efficient to pre-compute the index of these constant string attribute names, and set them by index.